cou/utils/openstack.py

```diff
@@ -147,3 +229,98 @@ def lookup(cls, component: str, version: str) -> List[str]:
             if version in version_range:
                 compatible_os_releases.append(openstack_release)
         return compatible_os_releases
+
+
+def determine_next_openstack_release(release: str) -> Tuple[str, str]:
+    """Determine the next release after the one passed as a str.
+
+    The returned value is a tuple of the form: ('2020.1', 'ussuri')
+
+    :param release: the release to use as the base
+    :type release: str
+    :returns: the release tuple immediately after the current one.
+    :rtype: Tuple[str, str]
+    :raises: ValueError if the current release doesn't actually exist
+    """
+    old_index = list(OPENSTACK_CODENAMES.values()).index(release)
+    new_index = old_index + 1
+    return list(OPENSTACK_CODENAMES.items())[new_index]
+
+
+class BasicStringComparator:
+    """Provides a class that will compare strings from an iterator type object.
+
+    Used to provide > and < comparisons on strings that may not necessarily be
+    alphanumerically ordered.  e.g. OpenStack or Ubuntu releases AFTER the
+    z-wrap.
+    """
+
+    _list: Optional[List] = None
+
+    def __init__(self, item):
+        """Do init."""
+        if self._list is None:
+            raise ValueError("Must define the _list in the class definition!")
+        try:
+            self.index = self._list.index(item)
+        except ValueError as exc:
+            raise ValueError(f"Item '{item}' is not in list '{self._list}'") from exc
+
+    def __eq__(self, other):
+        """Do equals."""
+        assert isinstance(other, (str, self.__class__))
+        return self.index == self._list.index(other)
+
+    def __ne__(self, other):
+        """Do not equals."""
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Do less than."""
+        assert isinstance(other, (str, self.__class__))
+        return self.index < self._list.index(other)
+
+    def __ge__(self, other):
+        """Do greater than or equal."""
+        return not self.__lt__(other)
+
+    def __gt__(self, other):
+        """Do greater than."""
+        assert isinstance(other, (str, self.__class__))
+        return self.index > self._list.index(other)
+
+    def __le__(self, other):
+        """Do less than or equals."""
+        return not self.__gt__(other)
+
+    def __repr__(self):
+        """Return the representation of CompareOpenStack."""
+        # pylint: disable=unsubscriptable-object
+        return f"{self.__class__.__name__}<{self._list[self.index]}>"
+
+    def __str__(self):
+        """Give back the item at the index.
+
+        This is so it can be used in comparisons like:
+
+        s_mitaka = CompareOpenStack('mitaka')
+        s_newton = CompareOpenstack('newton')
+
+        assert s_newton > s_mitaka
+
+        :returns: <string>
+        """
+        # pylint: disable=unsubscriptable-object
+        return self._list[self.index]
+
+
+class CompareOpenStack(BasicStringComparator):
```

Comment:
Why we are defining the `OPENSTACK_CODENAMES` as ordered dict and than never using it full potential? Why not defining our own [UserList](https://docs.python.org/3/library/collections.html#collections.UserList) or [UserDict](https://docs.python.org/3/library/collections.html#collections.UserList) object. Like this we can provide whole functionality in single object and not making `determine_next_openstack_release`, `BasicStringComparator` and `CompareOpenStack`.

We can do
```python
OPENSTACK_CODENAMES.get("xena").next_release()
OPENSTACK_CODENAMES["xena"].next_relase()

OPENSTACK_CODENAMES["xena"] > OPENSTACK_CODENAMES["yoga"]
```

Reply:
Good idea. I've added this kind of functionality in the new commit